hicommonwealth/commonwealth
# `[Launchpad Discovery]`: Multiple images per token idea draft

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8864

## Description of Changes
- Added loading state on image input when the image is being generated in quick token launch form
- Added multiple images state per token idea draft in quick token launch form
- Added micro UX/state cleanup and enhancements.

## "How We Fixed It"
N/A

## Test Plan
IMPORTANT: try to break it, we want to make sure that multiple state updates with deep objects/array and with timeouts are working correctly. Especially to test timeouts don't conflict with array/object mutations.

1. Enable `FLAG_TOKENIZED_COMMUNITY`
2. Visit the `/communities` page
3. Click on the "Launch token" button in the "Launch an idea" card and verify the drawer opens with the token launch form.
4. Click on the randomize button, and verify you see the form getting filled with a token idea with a typing animation.
5. Verify that you see loading animation on the image field as the image is being generated
6. Verify you can generate/upload multiple images and cycle through them
7. Generate 4 more ideas
8. Generate/upload multiple images per those 4 ideas
9. Verify when you cycle through each draft, you still see all the images you generated for that specific draft
10. Verify that when you move back to a specific draft, the last active image (b/w multiple images) on that draft is still the active image. Ex: Draft 1 has 5 images, and you have the 3rd image as the active one, when you cycle to another draft and come back to draft 1, It should display the image that you set as active, in this case, image 3.


## Deployment Plan
N/A

## Other Considerations
N/A
